afl/src/bin/cargo-afl.rs

```diff
@@ -224,21 +243,28 @@ where
     I: IntoIterator<Item = S>,
     S: AsRef<OsStr>,
 {
+    let no_sudo = env::var("NO_SUDO").is_ok();
     let cmd_path = common::afl_dir(None).join("bin").join(tool);
-    let mut cmd = Command::new(cmd_path);
+    let mut cmd = if !no_sudo && tool == "afl-system-config" {
+        let mut cmd = Command::new("sudo");
+        cmd.args([OsStr::new("--reset-timestamp"), cmd_path.as_os_str()]);
+        eprintln!("Running: {cmd:?}");
+        cmd
+    } else {
+        Command::new(cmd_path)
+    };
     cmd.args(args);
+
     let status = cmd.status().unwrap();
-    #[cfg(target_os = "macos")]
+
     if tool == "afl-fuzz" && !status.success() {
-        let sudo_cmd_path = common::afl_dir(None).join("bin").join("afl-system-config");
         eprintln!(
             "
 If you see an error message like `shmget() failed` above, try running the following command:
 
-    sudo {}
+    cargo-afl afl system-config
 
-Note: You will be prompted to enter your password.",
-            sudo_cmd_path.display()
+Note: You will be prompted to enter your password as root privileges are required and hence sudo is run within this command."
```

Comment:
```suggestion
Note: You might be prompted to enter your password as root privileges are required and hence sudo is run within this command."
```

FWIW, I added `--reset-timestamp` to the sudo invocation in attempt to address the "already used sudo" problem.

But I see your point about NOPASSWD.